tox.ini

```diff
@@ -50,16 +52,17 @@ deps =
     pytest
     ops
     pytest_operator
-    requests
     types-requests
     types-PyYAML
     pytest_asyncio
-    pydocstyle
+    pydocstyle>=2.10
+    ruff
 commands =
     pydocstyle {[vars]src_path}
     codespell {toxinidir}/. --skip {toxinidir}/.git --skip {toxinidir}/.tox \
       --skip {toxinidir}/build --skip {toxinidir}/lib --skip {toxinidir}/venv \
       --skip {toxinidir}/.mypy_cache --skip {toxinidir}/icon.svg
+    ruff {[vars]all_path}
```

Comment:
Do we want to keep the old tooling that provides overlapping functionality?

Reply:
I was planning on using both the tools for a little while just to see if anything is caught by the overlapping tooling that isn't caught by `ruff`. This should definitely be removed in the near future, in one of the next demos I'll discuss `ruff` a bit and we can decide whether we adopt it as a team. So far I haven't found anything that `ruff` doesn't catch, except the copyright notice which is a useful tool we would give up